test/java/controllers/KairosDbProxyControllerTest.java

```diff
@@ -194,4 +198,50 @@ public void testQueryRequiresAggregatorOff() {
         assertEquals(Http.Status.OK, result.status());
         assertEquals("{\"queries\":[]}", Helpers.contentAsString(result));
     }
+
+    @Test
+    public void testAddMergeAggregator() {
+        final Metric.Builder metric1Builder = new Metric.Builder()
+                .setName("metric1")
+                .setAggregators(ImmutableList.of(
+                        new Aggregator.Builder().setName("filter").build(),
+                        new Aggregator.Builder().setName("max").setSampling(
+                                new Sampling.Builder().setValue(1).setUnit(SamplingUnit.HOURS).build()).build()));
+        final Metric.Builder metric2Builder = new Metric.Builder()
+                .setName("metric2")
+                .setAggregators(ImmutableList.of(new Aggregator.Builder().setName("min").build()));
+        final MetricsQuery.Builder builder = new MetricsQuery.Builder()
+                .setStartTime(Instant.now())
+                .setMetrics(ImmutableList.of(metric1Builder.build(), metric2Builder.build()));
+
+        final Http.RequestBuilder request = Helpers.fakeRequest()
+                .method(Helpers.POST)
+                .uri("/api/v1/datapoints/query")
+                .header("Content-Type", "application/json")
+                .bodyJson(OBJECT_MAPPER.<JsonNode>valueToTree(builder.build()));
+
+        when(_mockKairosDbClient.queryMetrics(any())).thenReturn(
+                CompletableFuture.completedFuture(new MetricsQueryResponse.Builder()
+                        .setQueries(ImmutableList.of()).build())
+        );
+
+        // ***
+        // Test case where one metric doesn't have merge aggregator and is not one of "min,max,avg,count,sum"
+        // ***
+        final Result result = Helpers.invokeWithContext(request, Helpers.contextComponents(), () -> {
+            final CompletionStage<Result> completionStage = _controller.queryMetrics();
+            return completionStage.toCompletableFuture().get(10, TimeUnit.SECONDS);
+        });
+
+        assertEquals(Http.Status.OK, result.status());
+        assertEquals("{\"queries\":[]}", Helpers.contentAsString(result));
+
+        final MetricsQuery newMetricsQuery = _controller.checkAndAddMergeAggregator(builder.build());
+        assertEquals("merge", newMetricsQuery.getMetrics().get(0).getAggregators().get(0).getName());
+        assertEquals("filter", newMetricsQuery.getMetrics().get(0).getAggregators().get(1).getName());
+        assertEquals("max", newMetricsQuery.getMetrics().get(0).getAggregators().get(2).getName());
+        assertEquals(newMetricsQuery.getMetrics().get(0).getAggregators().get(2).getSampling(),
+                newMetricsQuery.getMetrics().get(0).getAggregators().get(0).getSampling());
+        assertEquals("min", newMetricsQuery.getMetrics().get(1).getAggregators().get(0).getName());
+    }
```

Comment:
This test seems to be doing two distinct things:
- testing the `checkAndAddMergeAggregator` method
- testing an HTTP request made against the controller

Could we split those into two test-methods? Having more granular tests makes it easier to figure out what's going wrong when something breaks.